### ISSUE TITLE
Connection issue

### ISSUE THREAD

**Issue by JohannaJacob** [greggman/HappyFunTimes#10]
I tried to use happyfuntimes on new computer, but can unfortunately not connect to happyfuntimes.

I get the error messages:

ERROR: parsing C:\Users\ Johanna Jacob\Documents\Game Jam\HappyFunTime_2\Assets\WebplayerTemplates\HappyFunTimes\package.json
ERROR: Reading C:\Users\ Johanna Jacob\Documents\Game Jam\HappyFunTime_2\Assets\WebplayerTemplates\HappyFunTimes\package.json
Type Error: Cannot read property 'gamesDir' of undefined

Any idea what can I do there?

**Comment by greggman** [comment]
Did you install it using the installer? Or did you clone the repo? If you cloned the repo you need to follow the Linux steps

http://docs.happyfuntimes.net/docs/linux.html

But leave off the sudo and the zenity stuff

**Comment by greggman** [comment]
The installer is at http://superhappyfuntimes.net/install

**Comment by JohannaJacob** [comment]
I am using windows.

I installed the happyfuntimes package from the unity assets store, but the game folder is a copy from my other pc.

**Comment by LDodson** [comment]
If you copied from another windows pc it could be a permissions issue.
Check the security settings on the folder properties.

**Comment by greggman** [comment]
There's a list of installed and added games in something like c:/users/username/appdata/local/happyfuntimes/installed-games.json 

If you copied from another computer then the paths inside that file are probably wrong. You can type

    hft remove --bad 

To remove all the games in that list with bad paths. You can install games by going to superhappyfuntimes.com and installing games. Or by typing

    hft install pathtogame.zip

where pathtogame.zip is a file from GitHub releases, not the bare repo zip

You can add an external game by CDing to the folder the game is in and typing

    hft add

Not sure how else to help. You can uninstall happyfuntimes, then You can delete c:/users/username/appdata/local/happyfuntimes and c:/users/username/appdata/local/greggman and then reinstalling

**Comment by greggman** [comment]
For unity of should just get it from the asset store

https://www.assetstore.unity3d.com/en/#!/content/19668

**Comment by JohannaJacob** [comment]
I removed the project and happyfuntimes. I re-imported the happyfuntimes pack from the asset store and re-installed the installer.

I tried to get one of the sample projects running, but I have the same problem here. :(

The path which the program tries to access is correct. I found the package.json there. But the computer does not...

**Comment by formula1** [comment]
It's likely not a network error as it were trying to access a non-existent property on a nonexistent config. I don't know how recent the release on the unity store was but I would assume its related to somewhere around [here](https://github.com/greggman/HappyFunTimes/blob/6c4e5d0e1978c1c74c56b02ad0941b1d3d88aecd/management/install.js#L137)

If you look at [getConfig](https://github.com/greggman/HappyFunTimes/blob/5b30a8cbeb5ee0036a1b9d7b515abae03a9e1a83/lib/config.js#L131) there's only one place we're returning undefined, around line 140. Interestingly enough that is also the only place where the caught error doesn't get rethrown.

If that's where the error is, everything would make sense

**Comment by JohannaJacob** [comment]
"If that's where the error is" --> how do I find out if this is the error?

Strange thing is that the project runs on my desktop pc (windows 7), my fioncés desktop pc (windows 10) and his notebook (windows 7), but not on my new notebook (windows 7). So it seems to be an issue with my notebook.

**Comment by greggman** [comment]
I will look into it as soon as I get home in a few hours

**Comment by formula1** [comment]
@JohannaJacob 

* find the Folder Happy Fun Times is installed in
* Look for the file `lib/config.js`
* go to line 140
* right above `return;` add the lines `console.log('we have an error')` and `console.log(e)` making it look like 
```
console.log('cannot read from configPath: ', g.configPath)
console.log(e)
return;
```
* Run it

**Comment by greggman** [comment]
Hmmm, I'm not having any luck reproducing your issue. I thought maybe it was because you have a space in your user name so I tried creating a new user with a space in their name but it still worked.

Can you think of anything else? What version of Windows? 7? 8? 10? Which version 32bit or 64bit?

**Comment by greggman** [comment]
On other idea. Saying that it's getting an error 

```
ERROR: Reading C:\Users\ Johanna Jacob\Documents\Game Jam\HappyFunTime_2\Assets\WebplayerTemplates\HappyFunTimes\package.json
Type Error: Cannot read property 'gamesDir' of undefined
```

Suggests it's not installed correctly.  If you open a command prompt and type 

    hft version

what does it say? For example mine says

<img width="334" alt="hft-cmd-prompt-version" src="https://cloud.githubusercontent.com/assets/234804/9425620/4a88ec20-490e-11e5-8d7d-ee5d70cdda4c.png">

if you get an error you can try 

    hft init

**Comment by JohannaJacob** [comment]
I am using windows 7 64 bit in german.

The hft version check says:

Microsoft Windows [Version 6.1.7601]
Copyright (c) 2009 Microsoft Corporation. Alle Rechte vorbehalten.

C:\Users\Johanna Jacob>hft version
hftVersion: 0.0.38
apiVersion: 1.14.0
C:\Users\Johanna Jacob>

**Comment by JohannaJacob** [comment]
@formula1 : hmm strange, I can neither find a "lib" folder nor a "config.js" file.

**Comment by formula1** [comment]
Just Opened up windows and reinstalled it... Alright, stick with me here (or wait for @greggman because he's the badass that made and fixes everything)

So my hft is in `C:\Users\Sam\AppData\Local\Greggman\HappyFunTimes`
Your's likely to reside in `C:\Users\Johanna Jacob\AppData\Local\Greggman\HappyFunTimes`

If it is not, then we have a different problem where we have to go to the path/environment variables to find out where it could possibly be :P (Hopefully we don't have to)

* There you will find `\lib`
* Inside there you will find `\config.js`
* Open it up in your favoirite text editor (I prefer [Notepad++](https://notepad-plus-plus.org/) on windows) 
* Go to line 140
* Edit away :)

**Comment by greggman** [comment]
`hft config`

will show you where the config file should be found

```

For example on mine it's

{
  "installDir": "/Users/gregg/src/happyfuntimes",
  "gamesDir": "/Users/gregg/Library/Application Support/HappyFunTimes/games",
  "options": [
    {
      "name": "dns",
      "value": false
    },
    {
      "name": "private",
      "value": false
    },
    {
      "name": "phonecontrol",
      "value": true
    }
  ],
  "configDir": "/Users/gregg/Library/Application Support/HappyFunTimes",
  "installedGamesListPath": "/Users/gregg/Library/Application Support/HappyFunTimes/installed-games.json"
}
```

but that part of happyfuntimes is working otherwise it would just say "happyfuntimes is not installed"


One more question. Inside your Unity project there should be a file, `Assets/WebPlayerTemplates/HappyFunTimes/package.json`. Does that file exist?

I guess I need you to walk me through each step you've taken. Sorry this is frustrating. I just have to be able to reproduce the bug to fix it.

**Comment by JohannaJacob** [comment]
Hi greggman, I checked the folder and the package.json file exists as well as the package.json.meta. Is there a new version of the programm available? I could try to deinstall and reinstall it again today.

**Comment by greggman** [comment]
I'm sorry this is so frustrating.

I tried making a new project on a separate drive, installing the plugin. I got no errors. It worked.

I then tried adding a space to the beginning of the folder name like you appear to have above. Your user folder is apparently `"<space>Johanna Jacob"`. That caused HFT to exit. I didn't get the same error as you but I did get an error. 

So, I upgraded HFT to use the latest version of node.js and that fixed the problem on my end.

Can you try HappyFunTimes version 0.0.42?

https://github.com/greggman/HappyFunTimes/releases/download/v0.0.42/HappyFunTimes-0.0.42-installer.exe

**Comment by JohannaJacob** [comment]
No problem greggman, bugs happen - I am happy that you give me support so patiently. :)

Unfortunately, the update did not fix my problem. :(
I can see no space in my user folder name. (The only one I see is the one between my first and surname - could this be a problem?
I think the update replaced only the files in my user folder, not the files in the project. Has there to be something updated too?

**Comment by greggman** [comment]
I tried switching windows into German mode but still no bug :(

Could you zip up your ``HappyFunTime_2` folder and send it to me?  github at greggman.com or put it dropbox or something and I can try to repo the bug here?

**Comment by JohannaJacob** [comment]
I fixed my problem! I renamed my User profile only to "Johanna" (incl. registry entry) instead of "Johanna Jacob" and it works now! I guess the space between first and surname was the problem.

I had to reinstall happyfuntimes afterwards, But now it works :D

If you still want to have a look:
https://www.dropbox.com/s/z16ptke7ytb8aq8/HappyFunTime_2.7z?dl=0

**Comment by greggman** [comment]
It's good to know that fixed your problem but unfortunately I tried spaces in names and was not able to reproduce the bug. I wish I could so I fix it for others. Well, at least it's working for you

**Comment by JohannaJacob** [comment]
Oh I thought you just had a problem with a space at the beginning of the user folder's name. Strange thing that it worked for me then o.o perhaps others will encounter no spacing problems when installing all for the first time. If you want I can create a second account on my machine to test spacing problems further.

**Comment by greggman** [comment]
No you're right. I had assumed it was a space anywhere would cause the bug but it's only a space in the username causes the bug. I've repoed it and am looking into it.

Thanks so much for your help finding this.

**Comment by JohannaJacob** [comment]
No problem, I enjoy hunting down bugs ^^

**Comment by greggman** [comment]
So, `hft init` would have fixed it.

I'm pretty sure the bug is in the installer. Looking into it

**Comment by greggman** [comment]
Ok, I think I fixed the installer in version 0.0.43. No need to test anything. I'll just cross my fingers. It worked for me.
